spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -621,6 +640,32 @@ public static void importSparkPartitions(
     }
   }
 
+  public static List<SparkPartition> filterPartitions(List<SparkPartition> partitions,
+                                                      Map<String, String> partitionFilter) {
+    if (partitionFilter.isEmpty()) {
+      // No partition filter arg
+
+      if (partitions.isEmpty()) {
+        throw new IllegalArgumentException("Cannot add files, no files found in the table.");
+      }
+      return partitions;
+    } else {
+      // Partition filter arg passed
+
+      List<SparkTableUtil.SparkPartition> filteredPartitions = partitions
+          .stream().filter(p -> p.getValues().entrySet().containsAll(partitionFilter.entrySet()))
+          .collect(Collectors.toList());
+
+      Preconditions.checkArgument(!filteredPartitions.isEmpty(),
```

Comment:
I feel like this validation should be outside too. It is specific to `add_files` while the method is used in multiple places.

Reply:
I can move them both to the import, I do think it should be an exception because I dislike silent noops